core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -85,28 +87,26 @@ oppia.factory('explorationData', [
           // changes applied. This makes a force-refresh necessary when changes
           // are discarded, otherwise the exploration-with-draft-changes
           // (which is cached here) will be reused.
-          return $http.get(explorationDataUrl, {
-            params: {
-              apply_draft: true
-            }
-          }).then(function(response) {
+          return EditableExplorationBackendApiService.fetchExploration(
+            explorationId, true).then(function(response) {
             $log.info('Retrieved exploration data.');
-            $log.info(response.data);
+            $log.info(response);
 
-            explorationData.data = response.data;
+            explorationData.data = response;
 
-            return response.data;
+            return response;
           });
         }
       },
       // Returns a promise supplying the last saved version for the current
       // exploration.
       getLastSavedData: function() {
-        return $http.get(explorationDataUrl).then(function(response) {
+        return ReadOnlyExplorationBackendApiService.loadLatestExploration(
```

Comment:
I believe this also needs to return Data as well. Surely the caller uses this data as well, no?